lib/engine/step/g_1846/track_and_token.rb

```diff
@@ -12,6 +12,13 @@ class TrackAndToken < TrackAndToken
         def buying_power(entity)
           @game.track_buying_power(entity)
         end
+
+        def lay_tile_action(action)
+          super
+          return unless @game.lake_shore_line && @upgraded && @game.class::LSL_HEXES.include?(action.hex.id)
```

Comment:
don't use unless with multiple conditions, switch it to an iff

Reply:
I had this: 
```
if @game.lake_shore_line && @upgraded && @game.class::LSL_HEXES.include?(action.hex.id)
            action.tile.icons.reject! { |icon| icon.name == @game.class::LSL_ICON }
end
  ```
  
but [the cops complained about guard clauses](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/GuardClause)

![Screenshot 2021-01-23 at 5 21 20 PM](https://user-images.githubusercontent.com/1711810/105618759-c4ed4780-5d9f-11eb-982c-20c165e29735.png)

Reply:
You can also use multiple returns

Reply:
yea, do an early return

Reply:
Hmm, I'm not sure what the expected patch is

Reply:
return super if !game.lsl || !@upgraded || ...

action.tile.icons.rject

Reply:
It looks like [the proposed diff](https://github.com/benjaminxscott/18xx/tree/1846_clear_lsl_test) results in 3 test fails

```
rspec './spec/assets_spec.rb[1:1:40:1]' # Assets #html 1846 22071 renders after_lay 451
rspec './spec/lib/engine/games/game_spec.rb[1:1:1]' # 18 Los Angeles hs_srwgrtvq_1602711223 matches result exactly
rspec './spec/lib/engine/games/game_spec.rb[16:1:1]' # 1846 22071 matches result exactly
```

It doesn't seem to result in the expected outcome, based on [a sample game](https://gist.github.com/benjaminxscott/aa25da23db6e496a0d2e99a104f27993)

![Screenshot 2021-01-25 at 1 02 11 PM](https://user-images.githubusercontent.com/1711810/105767229-a8701d00-5f0f-11eb-97ae-652e91deafff.png)

Reply:
so fix the logic? if ... but with the right conditions

Reply:
I should clarify, the link above is to a commit where I made that modification
(the `...` was just a shorthand, not being used literally)
https://github.com/benjaminxscott/18xx/commit/7ea16a9a0c639d9d27c6457444e44e0bab30862e

Reply:
yea because the logic is different

it should be

super
return if !x || !y

Reply:
OK I think I get it